include/nbl/video/decl/IBackendObject.h

```diff
@@ -25,13 +25,17 @@ class IBackendObject
         const ILogicalDevice* getOriginDevice() const;
 
         // to get useful debug messages and names in Renderdoc captures
-        virtual void setDebugName(const char* label) {} //=0;
+        virtual void setObjectDebugName(const char* label) const { m_debugName = label; }
+
+        const std::string& getObjectDebugName() const { return m_debugName; }
 
     protected:
         virtual ~IBackendObject();
 
     private:
         const core::smart_refctd_ptr<const ILogicalDevice> m_originDevice;
+
+        mutable std::string m_debugName;
```

Comment:
why do we even need to remember the debug name?

Reply:
256 is minimum it supports
but well ok

> why do we even need to remember the debug name?

because most GL resources are created in deferred fashion (pretty much everything that cant be shared between contexts, please read the code)

Reply:
> please read the code

yeah I read 1 hour later into the review and I was appalled by debug names being set on cached objects every time their state was flushed.